### ISSUE TITLE
Linking or copying consistent snapshots

### ISSUE THREAD

**Issue by vladimir-v-diaz** [theupdateframework/tuf#374]
A contributor reports:

> When making changes, always hard linking old versions to the current will result in all previous versions (1.root, 2.root, etc) pointing to the current (root.json).
> 
> Writing the consistent version and then linking the current to that (e.g. linking root.json -> 3.root.json) should work, but requires more changes because now "write current" doesn't happen in every case. Holding two copies is a small constant increase in storage (#versions -> #versions + 1)
> 
> Either solution is fine, I left the commented code to remind myself about the issue and clearly forgot to come back to it!

We probably want to support the second solution (for clients / devices with limited space). Creating a true copy or a hard link should also probably be an option (i.e., tuf.conf.consistent_method = copy (or link)).

**Comment by JustinCappos** [comment]
Did this get addressed?  We discussed this a long ways back.

**Comment by vladimir-v-diaz** [comment]
I partially implemented this here: https://github.com/vladimir-v-diaz/tuf/blob/ecordell-root-versioning/tuf/repository_lib.py#L2015-L2047.  The fix points `root.json` to specific consistent snapshots.

@kwang100 is working on the configurable option to either create a copy of the consistent snapshot, or a link to it.

Consistent snapshots are now written as follows:
```Bash
[2016-10-18 17:33:53,016 UTC] [tuf.repository_lib] [INFO] [write_metadata_file:2031@repository_lib.py]
Creating a consistent snapshot for u'/home/vlad/projects/tuf/tests/tmp6TEQkg/tmpL3SxRM/repository/metadata.staged/root.json'
                                                                                 
[2016-10-18 17:33:53,016 UTC] [tuf.repository_lib] [DEBUG] [write_metadata_file:2032@repository_lib.py]
Saving u'/home/vlad/projects/tuf/tests/tmp6TEQkg/tmpL3SxRM/repository/metadata.staged/2.root.json'
                                                                                 
[2016-10-18 17:33:53,022 UTC] [tuf.repository_lib] [INFO] [write_metadata_file:2041@repository_lib.py]
Pointing u'/home/vlad/projects/tuf/tests/tmp6TEQkg/tmpL3SxRM/repository/metadata.staged/root.json' to the consistent snapshot.
                                                                                 
[2016-10-18 17:33:53,023 UTC] [tuf.repository_lib] [INFO] [_write_compressed_metadata:2139@repository_lib.py]
Saving u'/home/vlad/projects/tuf/tests/tmp6TEQkg/tmpL3SxRM/repository/metadata.staged/2.root.json.gz'
                                                                                 
[2016-10-18 17:33:53,029 UTC] [tuf.repository_lib] [INFO] [write_metadata_file:2045@repository_lib.py]
Not creating a consistent snapshot for u'/home/vlad/projects/tuf/tests/tmp6TEQkg/tmpL3SxRM/repository/metadata.staged/snapshot.json'
...
```

**Comment by vladimir-v-diaz** [comment]
This issue has now been fully addressed.  Remaining feature (option to use either a hard link or file copy) merged in PR #385
